### PR TITLE
Milestone first hes been pushed and we're still working on couple of …

### PR DESCRIPTION
**Pull request by kshekh**
…things

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,13 +9,18 @@
...
+    "@ownego/polaris-vue": "^1.3.0",
     "@tailwindcss/forms": "^0.5.3",
     "@tailwindcss/line-clamp": "^0.4.2",
     "@tailwindcss/typography": "^0.5.8",
+    "core-js": "^3.8.3",
     "pinia": "^2.0.26",
     "postcss-cli": "^10.1.0",
     "postcss-nested": "^6.0.0",
     "postcss-nesting": "^10.2.0",
+    "remixicon": "^2.5.0",
```
Comment (giteshk11) [inline package.json:23]: Why do we need this icons package?

File: package.json [hunk 2]
```diff
@@ -25,10 +30,32 @@
...
+    },
+    "extends": [
+      "plugin:vue/vue3-essential",
+      "eslint:recommended"
+    ],
+    "parserOptions": {
+      "parser": "@babel/eslint-parser"
+    },
+    "rules": {}
+  },
```
Comment (giteshk11) [inline package.json:54]: It's better to have a separate config file.

File: package.json [hunk 3]
```diff
@@ -25,10 +30,32 @@
...
+  "browserslist": [
+    "> 1%",
+    "last 2 versions",
+    "not dead",
+    "not ie 11"
+  ]
```
Comment (giteshk11) [inline package.json:60]: It's better to have a separate config file.

File: src/App.vue [hunk 4]
```diff
@@ -1,85 +1,136 @@
...
+          </span>
+          <span v-else>{{post.desc}}</span>
+          <span class="read-more-label" v-if="countWords(post.desc) > 21" @click="post.readActivated = !post.readActivated">{{!post.readActivated ? 'Read More' : 'Read Less'}}</span>
+        </p>
+      </template>
+      <template v-else>
+        <p>{{post.desc}}</p>
+      </template>
+    </li>
+  </ul> -->
```
Comment (giteshk11) [inline src/App.vue:24]: Please remove this commented code if it's not used.

File: src/App.vue [hunk 5]
```diff
@@ -1,85 +1,136 @@
...
 </template>
 
-<style scoped>
-header {
-  line-height: 1.5;
-  max-height: 100vh;
+<script>
+// import HomeCustomer from './components/HomeCustomer.vue';
+// import CreateAccount from './components/CreateAccount.vue';
+// import Login from './components/Login.vue';
```
Comment (giteshk11) [inline src/App.vue:30]: Please remove this commented code,

File: src/App.vue [hunk 6]
```diff
@@ -1,85 +1,136 @@
...
+  -moz-osx-font-smoothing: grayscale;
   text-align: center;
-  margin-top: 2rem;
+  color: #2c3e50;
 }
-
-nav a.router-link-exact-active {
-  color: var(--color-text);
+.vue-logo {
+  width: 70px;
```
Comment (giteshk11) [inline src/App.vue:106]: No `px` values only use `rem`.

File: src/App.vue [hunk 7]
```diff
@@ -1,85 +1,136 @@
...
 }
-
-nav a.router-link-exact-active:hover {
-  background-color: transparent;
+.post-grid {
+  display: flex;
+  flex-wrap: wrap;
+  justify-content: center;
+  gap: 20px;
```
Comment (giteshk11) [inline src/App.vue:112]: No `px` values only use `rem`.

File: src/App.vue [hunk 8]
```diff
@@ -1,85 +1,136 @@
...
+  padding-left: 0;
 }
-
-nav a {
-  display: inline-block;
-  padding: 0 1rem;
-  border-left: 1px solid var(--color-border);
+.post-grid li {
+  border: 1px solid #ccc;
+  padding: 10px;
```
Comment (giteshk11) [inline src/App.vue:118]: No `px` values only use `rem`.

File: src/App.vue [hunk 9]
```diff
@@ -1,85 +1,136 @@
...
+  width: 300px;
+  border-radius: 4px;
+  box-shadow: 3px 3px 7px #ececec;
+  transition: all 0.27s ease-in-out;
 }
-
-nav a:first-of-type {
-  border: 0;
+.post-grid li:hover{
+  box-shadow: 3px 3px 7px #ccc;
```
Comment (giteshk11) [inline src/App.vue:125]: No `px` values only use `rem`.
Also, this can move into tailwind config so that it can be use later if required.

File: src/App.vue [hunk 10]
```diff
@@ -1,85 +1,136 @@
...
-    margin-top: 1rem;
-  }
+h3 {
+  text-transform: capitalize;
+}
+.read-more-label {
+   cursor: pointer;
+   font-weight: bold;
+   font-size: 14px;
+   margin-left: 10px;
```
Comment (giteshk11) [inline src/App.vue:134]: No `px` values only use `rem`.

File: src/components/CreateAccount.vue [hunk 11]
```diff
@@ -0,0 +1,136 @@
+<template>
+  <div class="flex flex-col min-h-screen">
+    <div class="flex-shrink-0 bg-white py-1.5 shadow relative z-30">
+      <div class="mx-auto w-full max-w-8xl px-2 sm:px-4 lg:px-5 lg:pr-14">
+        <router-link
+          to="/home"
+          class="relative px-2 sm:px-0 flex h-16 items-center justify-between"
+        >
+          <img src="../assets/logo.png" alt="image" class="mx-auto" />
```
Comment (giteshk11) [inline src/components/CreateAccount.vue:9]: Use the `relative` path instead of using `..`

File: src/components/CreateAccount.vue [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+<!-- Add "scoped" attribute to limit CSS to this component only -->
+<!-- <style lang="scss" scoped>
+$bg-gray : #DFDFDF;
+.bg-gray {
+  background-color: $bg-gray;
+}
+.create-account-wrapper {
+  background: $bg-gray;
+}
+</style> -->
```
Comment (giteshk11) [inline src/components/CreateAccount.vue:136]: Please remove this commented code if not required.

File: src/components/HomeCustomer.vue [hunk 13]
```diff
@@ -0,0 +1,299 @@
...
+                        </button>
+                      </div>
+                    </div>
+                  </a>
+                </li>
+              </ul>
+            </div>
+
+            <div class="">
+              <img src="../assets/setup-guide-thumb.svg" alt="setup-thumb" class="hidden xl:block" />
```
Comment (giteshk11) [inline src/components/HomeCustomer.vue:194]: Use `relative` path instead of `..`

File: src/components/Login.vue [hunk 14]
```diff
@@ -0,0 +1,167 @@
...
+<script>
+export default {
+  name: "LoginAccount",
+  components: {},
+};
+</script>
+
+<!-- Add "scoped" attribute to limit CSS to this component only -->
+<style lang="scss" scoped>
+$bg-gray: #dfdfdf;
```
Comment (giteshk11) [inline src/components/Login.vue:160]: This can be moved to tailwind config

File: src/components/TopNavbar.vue [hunk 15]
```diff
@@ -0,0 +1,109 @@
+<template>
+    <Disclosure as="header" class="bg-white shadow" v-slot="{ open }">
+      <div class="mx-auto max-w-7xl px-2 sm:px-4 lg:divide-y lg:divide-gray-200 lg:px-8">
+        <div class="relative flex h-16 justify-between">
+          <div class="relative z-10 flex px-2 lg:px-0">
+            <div class="flex flex-shrink-0 items-center">
+              <img class="h-8 w-auto" src="../assets/logo.png" alt="BrowTricks" />
```
Comment (giteshk11) [inline src/components/TopNavbar.vue:7]: Same here

File: src/components/TopNavbarLogin.vue [hunk 16]
```diff
@@ -0,0 +1,11 @@
+<template>
+   <div class="top-navbar bg-white drop-shadow-sm px-3 py-4">
+      <img src="../assets/logo.png" alt="image" class="mx-auto" />
```
Comment (giteshk11) [inline src/components/TopNavbarLogin.vue:3]: Same here

File: src/assets/global.scss [hunk 17]
```diff
@@ -32,6 +32,10 @@
...
  
-.kamran{
-    @apply bg-blue-500 #{!important}
+.step-list{
+    li:last-child{
+        .v-line{
+            @apply hidden
+        }
+    }
 }
```
Comment (giteshk11) [inline src/assets/global.scss:41]: Why is this file required?

File: src/main.js [hunk 18]
```diff
@@ -2,8 +2,8 @@ import { createApp } from 'vue'
 import { createPinia } from 'pinia'
 
 import App from './App.vue'
-import router from './router'
-import '../node_modules/typeface-inter/inter.css'
+import router from './router' 
+import 'typeface-inter'
```
Comment (giteshk11) [inline src/main.js:6]: Why are we loading font in js, instead of css?

File: tailwind.config.js [hunk 19]
```diff
@@ -66,6 +74,13 @@ module.exports = {
                 '4xl': '96rem',
             },
 
+            width:{
+                '3': '3px',
+                '68': '274px',
+                '119': '477px',
+                '22': '89px',
+            },
+
```
Comment (giteshk11) [inline tailwind.config.js:83]: No `px` values only use `rem`.

File: tailwind.config.js [hunk 20]
```diff
@@ -83,6 +98,7 @@ module.exports = {
             boxShadow: {
                 '1': '0px 9px 44px rgba(0, 0, 0, 0.06)',
                 '2': '0px 20px 41px rgba(0, 0, 0, 0.25)',
+                '3': '0px 1px 6px rgba(0, 0, 0, 0.1)',
```
Comment (giteshk11) [inline tailwind.config.js:101]: No `px` values only use `rem`.

File: tailwind.config.js [hunk 21]
```diff
@@ -96,10 +112,22 @@ module.exports = {
                 '5.5xl': '3.5rem',
                 '6.5xl': ['4rem', '1.10'],
                 '7.5xl': '5rem',
+                  '13'  : '13px',
+                  '15'  : '15px',
+                  '32'  : '32px',
```
Comment (giteshk11) [inline tailwind.config.js:117]: No `px` values only use `rem`.

File: tailwind.config.js [hunk 22]
```diff
@@ -96,10 +112,22 @@ module.exports = {
...
+            borderWidth:{
+              3:'3px',
+              5:'5px',
+              6:'6px',
+            },
+            ringWidth: {
+              3:'3px',
+              5:'5px',
+              6:'6px',
+            },
```
Comment (giteshk11) [inline tailwind.config.js:130]: No `px` values only use `rem`.

**Review by giteshk11 — CHANGES_REQUESTED** [review]
A couple of points
- I don't see any components which were mentioned as part of 1st issue i.e buttons, inputs, etc. We need those buttons and other elements as separate components as they will reduce code duplication.
- Please don't mix the style of APIs for vue, use only composition API.
- Please set up prettier to make your code look good and consistent.

**Comment by giteshk11** [comment]
Validations need to active only when a user has clicked inside the field.

File: package.json [hunk 23]
```diff
@@ -9,13 +9,18 @@
     "test:unit": "vitest --environment jsdom --root src/"
   },
   "dependencies": {
+    "@headlessui/vue": "^1.7.4",
+    "@heroicons/vue": "^2.0.13",
+    "@ownego/polaris-vue": "^1.3.0",
     "@tailwindcss/forms": "^0.5.3",
     "@tailwindcss/line-clamp": "^0.4.2",
     "@tailwindcss/typography": "^0.5.8",
```
Comment (giteshk11) [inline package.json:17]: These needs to be devdependency.

File: package.json (hunk 1 above)
Comment (kshekh) [reply package.json:23]: Separate components for Inputs, buttons, and other form elements are being created.
The mixing of styles of APIs has been removed.
Code has been formatted with "prettier"
"remixicon" has been removed and created icon as separate components

File: package.json (hunk 2 above)
Comment (kshekh) [reply package.json:54]: eslintConfig has been removed we don't use

File: package.json (hunk 3 above)
Comment (kshekh) [reply package.json:60]: browserslist: has been removed also.

File: src/App.vue (hunk 4 above)
Comment (kshekh) [reply src/App.vue:24]: Commented code has been removed from all the pages

File: src/App.vue (hunk 5 above)
Comment (kshekh) [reply src/App.vue:30]: Commented code has been removed from all the pages

File: src/main.js (hunk 18 above)
Comment (kshekh) [reply src/main.js:6]: We are using "npm i typeface" to import google fonts to store locally. You can read more about typeface here https://www.npmjs.com/package/typefaces

File: tailwind.config.js (hunk 19 above)
Comment (kshekh) [reply tailwind.config.js:83]: These values have been removed

File: tailwind.config.js (hunk 20 above)
Comment (kshekh) [reply tailwind.config.js:101]: These shadows are imported from Figma so the design impact should not be changed.
Tailwind CSS also used shadows in px instead of rem 
image

File: tailwind.config.js (hunk 21 above)
Comment (kshekh) [reply tailwind.config.js:117]: These values have been removed

**Comment by giteshk11** [comment]
I saw your comments, where is the updated code?

File: tailwind.config.js (hunk 22 above)
Comment (kshekh) [reply tailwind.config.js:130]: Ring width and Border width should be used in fixed format TailwindCSS also used border in px instead of rem. We added these values cause TailwindCSS does not have these defaults.
image

File: package.json (hunk 23 above)
Comment (kshekh) [reply package.json:17]: These have been moved

File: package.json [hunk 24]
```diff
@@ -9,13 +9,18 @@
...
+    "core-js": "^3.8.3",
     "pinia": "^2.0.26",
     "postcss-cli": "^10.1.0",
     "postcss-nested": "^6.0.0",
     "postcss-nesting": "^10.2.0",
+    "remixicon": "^2.5.0",
     "sass": "^1.56.1",
```
[the remark this answers is not in the document]
Comment (kshekh) [reply package.json:24]: We compile TailwindCSS from the .scss file and this file can be written with additional CSS if we will required something which can't control direct tailwind's utilities.
image

**Comment by giteshk11** [comment]
It seems like you are getting confused somewhere 
The issues you work on will have only one branch e.g. if you are working on issue no 1. it's going to have only one PR until all the reviews are completed and it's merged into the main branch

**Comment by kshekh** [comment]
Okay, I got it... Sorry for this one.
What should I need to do right now?

File: package.json (hunk 24 above)
[the remark this answers is not in the document]
Comment (giteshk11) [reply package.json:24]: That's fine please remove the all the .scss file and convert it to normal css file, also remove the node module as well.

**Comment by giteshk11** [comment]
That's ok for now, just make sure for next issues you don't do it this way.

Thanks

**Comment by kshekh** [comment]
Okay, I will aware and it will not be repeated next time.
Thanks

File: tailwind.config.js (hunk 22 above)
Comment (giteshk11) [reply tailwind.config.js:130]: that's fine you can covert using this https://nekocalc.com/px-to-rem-converter

File: tailwind.config.js (hunk 20 above)
Comment (giteshk11) [reply tailwind.config.js:101]: that's fine, please use this https://nekocalc.com/px-to-rem-converter to convert

File: src/main.js (hunk 18 above)
Comment (giteshk11) [reply src/main.js:6]: This package has not been updated since 2020.
Since CSS gives me an exclusive way to deal with fonts why should I use javascript to increase the overhead.
Also, we should not use too many npm packages as it affects the bundle size.

**Action Item**
Please download whatever font is used in the designs, import it using CSS and use it.

**Comment by joeelia** [comment]
I dont care that this has comments. Its too confusing to have multiple PRs. Deleting this.
